### PR TITLE
Move assert function to common.js file in test dir

### PR DESCRIPTION
**Pull request by theworkflow**
Pass message to assert functions in `test-nucleus-get-set-env.js`
Move assert function to common.js file in test dir

### CODE REVIEW

**Comment by creationix** [comment]
This shouldn't work.  The relative path just happens to work because I'm not protecting escaping from the bundle directory.  Can you move the common folder to inside the parallel folder?  Then use full paths to the file when using dofile.  `dofile("common/assert.js")`

When we use the custom main feature, the bundle root becomes the directory holding the custom main.

**Comment by Fishrock123** [comment]
@creationix It does work though. My other tests already use it - to access the `fixtures` directory from `parallel`. (They do not necessarily depend on it, but not having this features makes these things  very messy.)

Is there a reason we _must_ use a bundle directory? It seems that is only useful if we are running in zipped mode?

**Comment by Fishrock123** [comment]
@creationix Should we open a new issue to discuss requiring bundled mode when not using zip files in more detail?

**Comment by creationix** [comment]
@Fishrock123 that is a good question.  I suppose that custom main is already different behavior than running in zip mode.  I just don't want people getting the false idea that reading outside the bundle will continue to work when they zip their app.

**Comment by Fishrock123** [comment]
@ashleygwilliams Think the above may be solvable by clear, visible documentation? Or should we take some other action?

@creationix Could it be possible to make specialized messages for read-out-of-bundle when in "zipped mode"?

File: test/parallel/common/assert.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+function assert(thing, message) {
+  if (thing) return;
+  throw new Error(message);
+}
+
+return assert
```
Comment (Fishrock123) [inline test/parallel/common/assert.js]: @harlanj Would you mind to add a blank line to the end of the file? It helps git not have some legacy strangeness. :)

File: test/parallel/test-nucleus-get-set-env.js [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
-}
+var assert = nucleus.dofile("./common/assert.js")
 
 // get non-existing
-assert(nucleus.getenv('__NUCLEUS_TEST') === null)
+assert(nucleus.getenv('__NUCLEUS_TEST') === null, 'nonexisting env var should be `null`')
 
 // set new
-assert(nucleus.setenv('__NUCLEUS_TEST', 'nice day @nodeconf') === undefined)
+assert(nucleus.setenv('__NUCLEUS_TEST', 'nice day @nodeconf') === undefined, 'created env should return `undefined`')
```
Comment (Fishrock123) [inline test/parallel/test-nucleus-get-set-env.js]: `setenv should ...`

File: test/parallel/test-nucleus-get-set-env.js [hunk 3]
```diff
@@ -1,36 +1,31 @@
...
-assert(nucleus.getenv('__NUCLEUS_TEST') === 'nice day @nodeconf')
+assert(nucleus.getenv('__NUCLEUS_TEST') === 'nice day @nodeconf', 'get existing env should return value')
 
 // newly set in envkeys
-assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') !== -1)
+assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') !== -1, 'existing key index should not equal -1')
 
 // re-set
-assert(nucleus.setenv('__NUCLEUS_TEST', 'hi there') === undefined)
+assert(nucleus.setenv('__NUCLEUS_TEST', 'hi there') === undefined, 'setting existing env should return `undefined`')
```
Comment (Fishrock123) [inline test/parallel/test-nucleus-get-set-env.js]: ditto about the `setenv` comment

File: test/parallel/test-nucleus-get-set-env.js [hunk 4]
```diff
@@ -1,36 +1,31 @@
...
-assert(nucleus.setenv('__NUCLEUS_TEST', 'nice day @nodeconf') === undefined)
+assert(nucleus.setenv('__NUCLEUS_TEST', 'nice day @nodeconf') === undefined, 'created env should return `undefined`')
 
 // get newly set
-assert(nucleus.getenv('__NUCLEUS_TEST') === 'nice day @nodeconf')
+assert(nucleus.getenv('__NUCLEUS_TEST') === 'nice day @nodeconf', 'get existing env should return value')
 
 // newly set in envkeys
-assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') !== -1)
+assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') !== -1, 'existing key index should not equal -1')
```
Comment (Fishrock123) [inline test/parallel/test-nucleus-get-set-env.js]: `env variable should be set in envkeys`

File: test/parallel/test-nucleus-get-set-env.js [hunk 5]
```diff
@@ -1,36 +1,31 @@
...
-assert(nucleus.setenv('__NUCLEUS_TEST', 'hi there') === undefined)
+assert(nucleus.setenv('__NUCLEUS_TEST', 'hi there') === undefined, 'setting existing env should return `undefined`')
 
 // get re-set
-assert(nucleus.getenv('__NUCLEUS_TEST') === 'hi there')
+assert(nucleus.getenv('__NUCLEUS_TEST') === 'hi there', 'get existing key should return `hi there`')
 
 // re-set in envkeys
-assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') !== -1)
+assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') !== -1, 'existing key index should not equal -1')
```
Comment (Fishrock123) [inline test/parallel/test-nucleus-get-set-env.js]: ditto about `envkeys`

File: test/parallel/test-nucleus-get-set-env.js [hunk 6]
```diff
@@ -1,36 +1,31 @@
...
 
 // unset
-assert(nucleus.unsetenv('__NUCLEUS_TEST') === undefined)
+assert(nucleus.unsetenv('__NUCLEUS_TEST') === undefined, 'unsetting env should return `undefined`')
```
Comment (Fishrock123) [inline test/parallel/test-nucleus-get-set-env.js]: similar to `setenv`

File: test/parallel/test-nucleus-get-set-env.js [hunk 7]
```diff
@@ -1,36 +1,31 @@
...
 
 // get now non-existing
-assert(nucleus.getenv('__NUCLEUS_TEST') === null)
+assert(nucleus.getenv('__NUCLEUS_TEST') === null, 'nonexisting env var should be `null`')
 
 // no longer in envkeys
-assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') === -1)
+assert(nucleus.envkeys().indexOf('__NUCLEUS_TEST') === -1, 'nonexisting key index should equal -1')
```
Comment (Fishrock123) [inline test/parallel/test-nucleus-get-set-env.js]: also `envkeys`

**Comment by Fishrock123** [comment]
@harlanj looks good on the code side minus some minor things!

**Comment by Fishrock123** [comment]
Might hold this a day to settle https://github.com/creationix/nucleus/issues/7 (license silliness for future-proofing)

**Comment by creationix** [comment]
>  I just don't want people getting the false idea that reading outside the bundle will continue to work when they zip their app.

Funny story. Today at work we had an incident with the luvit agent.  There was a bug where a path had one too many dots ('..' vs '.') and was reading a file outside the bundle root.  It passed all the tests since the tests ran in luvi mode, but when the agent was deployed, it crashed.

So my fear of differing behavior causing trouble wasn't unfounded after all.

I'm fine, however, with adding an explicit flag or something that allows overriding the root path.  Or maybe we can have it look for a `main.js` in parent directories and automatically finding the app root.

**Comment by Fishrock123** [comment]
I've been pretty busy. Will open a new issue for this discussion tomorrow.

**Comment by Fishrock123** [comment]
Looks good, just pending other discussions.

**Comment by theworkflow** [comment]
Just wanted to check and see if there was any update on this @creationix @Fishrock123

**Comment by Fishrock123** [comment]
Sorry, I think it should be merge-able. In the future we'll just run the tests with something like the rustyduk implementation's `--no-bundle`.

I'll try to take a full look when I can! 

(p.s. There are lots of good issues that could be picked up in rustyduk if you are interested in JS & Rust: https://github.com/nucleus-js/rustyduk/issues)

**Comment by creationix** [comment]
I'm fine merging this now if @Fishrock123 is ok with it.

**Comment by Fishrock123** [comment]
yeah looks good to me

I still need to get rustyduk on these days but that is for a different thread..

**Comment by theworkflow** [comment]
Cool. Thanks guys!

**Pull request merged**